core/api/views/business_plan.py

```diff
@@ -459,33 +458,32 @@ def get_queryset(self):
         return queryset
 
     def get_wb(self, method):
+        year_start = int(self.request.query_params.get("year_start"))
+        year_end = int(self.request.query_params.get("year_end"))
+        agency_id = self.request.query_params.get("agency_id")
+
         # get all activities between year_start and year_end
         queryset = self.filter_queryset(self.get_queryset())
 
         data = BPActivityExportSerializer(queryset, many=True).data
 
-        limits = queryset.aggregate(
-            min_year=Min("values__year"), max_year=Max("values__year")
-        )
-
-        if start_year := int(self.request.query_params.get("year_start", "0")):
-            # If there is no data, or only partial data. Ensure we have fields for
-            # start_year, start_year + 1, start_year + 2, after start_year + 2
-            limits["min_year"] = limits["min_year"] or start_year
-            limits["max_year"] = max(limits["max_year"] or -math.inf, start_year + 3)
-
         wb = openpyxl.Workbook()
         sheet = wb.active
         sheet.title = "Business Plans"
         configure_sheet_print(sheet, sheet.ORIENTATION_LANDSCAPE)
 
         BusinessPlanWriter(
             sheet,
-            min_year=limits["min_year"],
-            max_year=limits["max_year"],
+            min_year=year_start,
+            max_year=year_end + 1,
         ).write(data)
 
-        name = f"Business Plans {limits['min_year']}-{limits['max_year'] - 1}"
+        if agency_id:
+            agency = get_object_or_404(Agency, id=agency_id)
```

Comment:
This check should be performed at the beginning of the method to ensure no operations are executed if the agency does not exist.